LSL/liblsl/CMakeLists.txt

```diff
@@ -1,36 +1,165 @@
-SET( CMAKE_LIBRARY_OUTPUT_DIRECTORY_RELEASE ${CMAKE_CURRENT_LIST_DIR}/bin )
-SET( CMAKE_RUNTIME_OUTPUT_DIRECTORY         ${CMAKE_CURRENT_LIST_DIR}/bin )
-SET( CMAKE_RUNTIME_OUTPUT_DIRECTORY_DEBUG   ${CMAKE_CURRENT_LIST_DIR}/bin )
-SET( CMAKE_RUNTIME_OUTPUT_DIRECTORY_RELEASE ${CMAKE_CURRENT_LIST_DIR}/bin )
 
 # Allow to use either the operating system supplied boost distribution or the shrink-wrapped one in the external/ directory
-option (USE_SYSTEM_BOOST "Use system Boost libraries instead of the version shipped with LSL." OFF)
+option (LSL_USE_SYSTEM_BOOST "Use system Boost libraries instead of the version shipped with LSL." OFF)
+
+# Build static library
+option (LSL_BUILD_STATIC "Also build static LSL library." OFF)
+
+set (sources
+	src/api_config.cpp
+	src/api_config.h
+	src/cancellable_streambuf.h
+	src/cancellation.h
+	src/common.cpp
+	src/common.h
+	src/consumer_queue.cpp
+	src/consumer_queue.h
+	src/data_receiver.cpp
+	src/data_receiver.h
+	src/dllmain.cpp
+	src/endian/conversion.hpp
+	src/endian/detail/intrinsic.hpp
+	src/info_receiver.cpp
+	src/info_receiver.h
+	src/inlet_connection.cpp
+	src/inlet_connection.h
+	src/legacy/legacy_abi.cpp
+	src/legacy/legacy_abi.h
+	src/lsl_continuous_resolver_c.cpp
+	src/lsl_freefuncs_c.cpp
+	src/lsl_inlet_c.cpp
+	src/lsl_outlet_c.cpp
+	src/lsl_streaminfo_c.cpp
+	src/lsl_xml_element_c.cpp
+	src/portable_archive/portable_archive_exception.hpp
+	src/portable_archive/portable_iarchive.hpp
+	src/portable_archive/portable_oarchive.hpp
+	src/pugixml/pugiconfig.hpp
+	src/pugixml/pugixml.cpp
+	src/pugixml/pugixml.hpp
+	src/resolver_impl.cpp
+	src/resolver_impl.h
+	src/resolve_attempt_udp.cpp
+	src/resolve_attempt_udp.h
+	src/sample.cpp
+	src/sample.h
+	src/send_buffer.cpp
+	src/send_buffer.h
+	src/socket_utils.cpp
+	src/socket_utils.h
+	src/stream_info_impl.cpp
+	src/stream_info_impl.h
+	src/stream_inlet_impl.h
+	src/stream_outlet_impl.cpp
+	src/stream_outlet_impl.h
+	src/tcp_server.cpp
+	src/tcp_server.h
+	src/time_postprocessor.cpp
+	src/time_postprocessor.h
+	src/time_receiver.cpp
+	src/time_receiver.h
+	src/udp_server.cpp
+	src/udp_server.h
+)
+set(headers
+	include/lsl_c.h
+	include/lsl_cpp.h
+)
+
+set(target lsl)
+if (CMAKE_SIZEOF_VOID_P EQUAL 8)
+	set(lslplatform 64)
+else()
+	set(lslplatform 32)
+endif()
+set(lslplatform ${lslplatform}_${CMAKE_CXX_COMPILER_ID}${MSVC_VERSION})
 
-# Build shared and/or static library
-option (BUILD_SHARED "Build shared library." ON)
-option (BUILD_STATIC "Build static library." OFF)
+set(lsl_export_targets ${target})
 
-if (USE_SYSTEM_BOOST)
-  find_package(Boost COMPONENTS chrono filesystem serialization system thread REQUIRED)
+if (LSL_USE_SYSTEM_BOOST)
+	find_package(Boost COMPONENTS chrono filesystem serialization system thread REQUIRED)
+	set(LSL_BOOST_LIBS Boost::filesystem Boost::serialization Boost::thread)
 else ()
-  add_subdirectory (external)
-endif (USE_SYSTEM_BOOST)
+	add_subdirectory (external)
+	set(LSL_BOOST_LIBS "lslboost")
```

Comment:
The library target lslboost is declared in the directory. Linking to it sets the required properties (include paths, compiler defines, etc.)